lit_nlp/client/modules/perturbations_table_module.ts

```diff
@@ -0,0 +1,310 @@
+        ${this.renderTable(source, tableRows, deltaRowsById)}
+      </div>
+    `;
+  }
+
+  private renderHeader(generationsCount: number, rowsCount: number, sourceIndex: number) {
+    const toggleFilterSelected = () => {
+      this.filterSelected = !this.filterSelected;
+    };
+    return html`
+      <div class="info">
+        <div class="header-and-actions">
+          <div class="header-text">
+            Generated ${rowsCount === 1 ? '1 datapoint' : `${rowsCount} datapoints`}
+            from ${generationsCount === 1 ? '1 perturbation' : `${generationsCount} perturbations`}
+          </div>
+          <lit-checkbox
+            label="Only show if selected"
+            ?checked=${this.filterSelected}
+            @change=${toggleFilterSelected}
+          ></lit-checkbox>
+        </div>
+        ${this.renderNavigationStrip(sourceIndex)}
+       </div>
+    `;
+  }
+
+  private renderNavigationStrip(sourceIndex: number) {
+    const sources = this.deltasService.sourcesForModel(this.model);
+    if (sources.length === 1) {
+      return null;
+    }
+
+    const onChangeOffset = (delta: number) => {
+      const infos = this.shadowRoot!.querySelectorAll('.info');
+      const nextIndex = sourceIndex + delta;
+      if (nextIndex < infos.length && nextIndex >= 0) {
+        this.lastSelectedSourceIndex = nextIndex;
+       }
+    };
+
+    const previousButton = html`
+      <mwc-icon class='icon-button'
+        @click=${() => {onChangeOffset(-1);}}>
+        chevron_left
+      </mwc-icon>
+    `;
+    const nextButton = html`
+      <mwc-icon class='icon-button'
+        @click=${() => {onChangeOffset(1);}}>
+        chevron_right
+      </mwc-icon>
+    `;
+    const placeholderButton = html`<div class="icon-placeholder"> </div>`;
+    return html`
+      <span class="navigation-strip">
+        ${sourceIndex + 1} of ${sources.length} fields
+        <span class="navigation-buttons">
+          ${sourceIndex - 1 >= 0 ? previousButton : placeholderButton}
+          ${sourceIndex + 1 < sources.length ? nextButton : placeholderButton}
+        </span>
+      </span>
+    `;
+  }
+
+  /* Enforce selection */
+  private filteredDeltaRows(deltaRows: DeltaRow[]): DeltaRow[] {
+    return deltaRows.filter(deltaRow => {
+      return (this.selectionService.isIdSelected(deltaRow.d.id) ||
+        this.selectionService.isIdSelected(deltaRow.parent.id));
+    });
+  }
+
+  private formattedTableRows(deltaRows: DeltaRow[]): TableData[] {
+    const BLANK = '-';
+    const meanAbsDelta = d3.mean(deltaRows.filter(d => d.delta != null), d => {
+      return Math.abs(d.delta!);
+    });
+
+    // The id column is hidden, but used for lookups.
+    return deltaRows.map((deltaRow: DeltaRow) => {
+      const {before, after, delta, d, parent}  = deltaRow;
+      const row: TableData = [
+        d.id, // ID_COLUMN
+        this.renderSentenceWithDiff(parent.data.sentence, d.data.sentence), // SENTENCE_COLUMN
```

Comment:
hardcoded to "sentence" input field?

Reply:
👍 Yep, this needs to be generalized and I'll take another look at the UX for that (eg, all input text fields, maybe add a button for selecting columns similar to the data table if we expect >2 text columns).

Reply:
Thanks!  This is generalized to any text input now, and will show multiple columns.